src/libcrun/linux.c

```diff
@@ -1611,7 +1611,16 @@ libcrun_create_dev (libcrun_container_t *container, int devfd, int srcfd,
   int rootfsfd = get_private_data (container)->rootfsfd;
   const char *rootfs = get_private_data (container)->rootfs;
   size_t rootfs_len = get_private_data (container)->rootfs_len;
-  const char *rel_dev = relative_path_under_dev (device->path);
+  // Normalize path by removing trailing slash if it exists.
+  cleanup_free char *normalized_path = xstrdup (device->path);
+  size_t path_len = strlen (device->path);
+  memcpy (normalized_path, device->path, path_len);
+
+  while (path_len > 0 && normalized_path[path_len - 1] == '/')
+    {
+      normalized_path[--path_len] = '\0';
+    }
```

Comment:
No need to open-code `strrchr()`, almost.

Reply:
If we have to handle double-slashes at the end per:

- https://github.com/containers/crun/pull/1517#discussion_r1714368759

Then, this might need to be moved into a utility function we can reuse, perhaps.